src/hnsw/build.c

```diff
@@ -462,7 +473,10 @@ static void BuildIndex(
         }
 
         UpdateProgress(PROGRESS_CREATEIDX_PHASE, PROGRESS_HNSW_PHASE_IN_MEMORY_INSERT);
-        LanternBench("build hnsw index", ScanTable(buildstate));
+
+        if(buildstate->heap != NULL) {
+            LanternBench("build hnsw index", ScanTable(buildstate));
```

Comment:
Does this not run for unlogged tables?
If it does not run, how is the index structure populated?

Reply:
I don't think ScanTable can run for unlogged tables because it calls table_index_build_scan, which accesses data in buildstate->heap (buildstate->heap references NULL in the case of unlogged tables, leading to a runtime exception when accessed). Instead, I populated the IndexInfo struct by calling BuildIndexInfo at the beginning of ldb_ambuildunlogged.